lib/puppet/provider/cert/katello_ssl_tool.rb

```diff
@@ -15,7 +15,7 @@ def generate!
       FileUtils.mkdir_p(build_path)
       FileUtils.cp(resource[:custom_pubkey], build_path(File.basename(pubkey)))
       FileUtils.cp(resource[:custom_privkey], build_path(File.basename(privkey)))
-      FileUtils.cp(resource[:custom_req], build_path(File.basename(req_file)))
+      FileUtils.cp(resource[:custom_req], build_path(File.basename(req_file))) if resource[:custom_req]
```

Comment:
Granted, looking back at my own comments, I am not sure this even works without deeper testing. But, I think this might fail due to line 10 above.

Reply:
Good one. I added an extra condition but I hadn't reached this point in the code yet to write some proper tests so I'm relying on @ahumbe or @tbrisker to test this.